File: 0001-two-sum/0001-two-sum.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        int n=nums.size();
        unordered_map<int,int>m;
        vector<int> res;
        for(int i=0;i<n;i++)
        {
            if(m.count(target-nums[i])==1)
            {
                res.push_back(i);
                res.push_back(m[target-nums[i]]);
                return res;

            }
            m[nums[i]]=i;
        }
        return res;
    }
};

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        HashMap<Integer,Integer> res=new HashMap<>();
        
        for(int i=0;i<nums.length;i++)
        {
            if(res.containsKey(nums[i]))
            {
                return new int[]{i,res.get(nums[i])};
            }
            res.put(target-nums[i],i);
        }
        return new int[]{};
    }
}